execution/types/runtime_decoder_test.go

```diff
@@ -0,0 +1,344 @@
+package types
+
+import (
+	"bytes"
+	"strconv"
+	"testing"
+
+	"github.com/ChainSafe/gossamer/lib/common"
+	sc "github.com/LimeChain/goscale"
+	"github.com/LimeChain/gosemble/mocks"
+	primitives "github.com/LimeChain/gosemble/primitives/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+)
+
+const (
+	moduleOneIdx           = sc.U8(0)
+	functionIdx            = sc.U8(0)
+	signedExtrinsicVersion = 132
+)
+
+var (
+	signedExtrinsicBytes = []byte{
+		byte(signedExtrinsicVersion),                                                                      // version
+		0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, // signer
+		0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+		1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+		1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+		1, 1, 1, 1, 1, 1, 1, // signature,
+		// extra
+		uint8(moduleOneIdx), uint8(functionIdx), // call
+	}
+
+	header = primitives.Header{
+		ParentHash: primitives.Blake2bHash{
+			FixedSequence: sc.BytesToFixedSequenceU8(parentHash)},
+		Number:         5,
+		StateRoot:      primitives.H256{FixedSequence: sc.BytesToFixedSequenceU8(stateRoot)},
+		ExtrinsicsRoot: primitives.H256{FixedSequence: sc.BytesToFixedSequenceU8(extrinsicsRoot)},
+		Digest:         primitives.Digest{},
+	}
+
+	parentHash      = common.MustHexToHash("0x3aa96b0149b6ca3688878bdbd19464448624136398e3ce45b9e755d3ab61355c").ToBytes()
+	stateRoot       = common.MustHexToHash("0x3aa96b0149b6ca3688878bdbd19464448624136398e3ce45b9e755d3ab61355b").ToBytes()
+	extrinsicsRoot  = common.MustHexToHash("0x3aa96b0149b6ca3688878bdbd19464448624136398e3ce45b9e755d3ab61355a").ToBytes()
+	moduleFunctions = map[sc.U8]primitives.Call{}
+)
+
+var (
+	mockModuleOne *mocks.Module
+	mockCallOne   *mocks.Call
+)
+
+func Test_RuntimeDecoder_New(t *testing.T) {
+	target := setupRuntimeDecoder()
+	expect := runtimeDecoder{
+		modules: []primitives.Module{mockModuleOne},
+		extra:   mockSignedExtra,
+	}
+
+	assert.Equal(t, expect, target)
+}
+
+func Test_RuntimeDecoder_DecodeBlock_ZeroExtrinsicsEmptyBody(t *testing.T) {
+	target := setupRuntimeDecoder()
+
+	lenExtrinsics := sc.ToCompact(0).Bytes()
+	buff := bytes.NewBuffer(append(header.Bytes(), lenExtrinsics...))
+
+	resultBlock := target.DecodeBlock(buff)
+
+	expectedBlock := NewBlock(header, sc.Sequence[primitives.UncheckedExtrinsic]{})
+
+	assert.Equal(t, expectedBlock, resultBlock)
+}
+
+func Test_RuntimeDecoder_DecodeBlock_Module_Not_Exists(t *testing.T) {
```

Comment:
I think that this test has to be for DecodeCall.

Reply:
Fixed